scp/src/main/java/org/motechproject/nms/scp/service/impl/SecureCopyServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+package org.motechproject.nms.scp.service.impl;
+
+import org.motechproject.nms.scp.service.exception.SecureCopyException;
+import org.motechproject.nms.scp.service.exception.SortException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+
+@Service("secureCopyService")
+public final class SecureCopyServiceImpl {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecureCopyServiceImpl.class);
+
+
+    private static String getStderr(Process p) {
+        StringBuilder sb = new StringBuilder();
+        BufferedReader b = new BufferedReader(new InputStreamReader(p.getErrorStream()));
+        String line;
+
+        try {
+            while ((line = b.readLine()) != null) {
+                sb.append(line);
+                sb.append('\n');
+            }
+
+            b.close();
+        } catch (IOException e) {
+            LOGGER.error("Error trying to retrieve scp error message: {}", e.getMessage());
+        }
+        return sb.toString();
+    }
+
+
+    public void copyFrom(String user, String host, String identityFile, String remoteSource,
+                                String localDestination) throws SecureCopyException {
+
+        Runtime r = Runtime.getRuntime();
+        try {
+            String identity;
+            if (identityFile == null || identityFile.isEmpty()) {
+                identity = "";
+            } else {
+                identity = String.format("-i %s ", identityFile);
+            }
+            String command = String.format("scp %s%s@%s:%s %s", identity, user, host, remoteSource, localDestination);
```

Comment:
Same comment as above.

Plus the additional comment of why is this code in both the imi and scp modules?